packages/ts-morph/src/compiler/types/Type.ts

```diff
@@ -233,9 +233,19 @@ export class Type<TType extends ts.Type = ts.Type> {
     }
 
     /**
-     * Gets the target type of a type reference if it exists.
+     * Gets the reference type if this type is an instantiated type.
      *
-     * For example, given `Promise<string>` this will return `Promise<T>`.
+     * Specifically:
+     *
+     * - If this type is an instantiated type then returns the reference type.
+     * - If this type is a reference type then returns itself (same reference).
+     * - Otherwise returns `undefined`.
+     *
+     * For example:
+     *
+     * - Given `Promise<string>` this will return `Promise<T>`.
+     * - Given `Promise<T>` this will return the same `Promise<T>`.
+     * - Given `string` this will return `undefined`.
      */
```

Comment:
I think a runnable `@example` block could be great. Something like this for example:

```ts
const exps = new tsm.Project({ useInMemoryFileSystem: true })
  .createSourceFile(
    'example.ts',
    `
      export type Generic<T> = {}
      export const foo: Generic<string> = {}
      export const bar: string = ''
    `
  )
  .getExportedDeclarations()

exps.get('Generic')[0].getType().getTargetType() // ?
exps.get('foo')[0].getType().getTargetType() === exps.get('Generic')[0].getType()
exps.get('bar')[0].getType().getTargetType() === undefined
```

Note for some reason the exported `Generic` target type is not returning self right now, but rather `undefined` which is not what I expected. will look into it later.

Do you think this or something like it would be a good idea and helpful for users?

Reply:
Thanks for the suggestion, but I'd rather not at this moment. Updating examples for every scenario becomes too much to maintain. I think they will get the idea from the `Promise<string>` -> `Promise<T>` example in the "For example" section above.